src/providers/flagsView.ts

```diff
@@ -57,31 +57,38 @@ export class LaunchDarklyTreeViewProvider implements vscode.TreeDataProvider<Fla
 	}
 
 	getChildren(element?: FlagNode): Thenable<FlagNode[]> {
-		if (!this.flagNodes) {
+		if (typeof this.flagNodes === 'undefined' || this.flagNodes.length == 0) {
 			return Promise.resolve([new FlagNode(this.ctx, 'No Flags Found.', NON_COLLAPSED)]);
 		}
 
 		return Promise.resolve(element ? element.children : this.flagNodes);
 	}
 
+	setFlagsStore(flagstore: FlagStore): void {
+		this.flagStore = flagstore;
+		this.flagUpdateListener();
+	}
+
 	async getFlags(): Promise<void> {
 		try {
-			const flags = await this.api.getFeatureFlags(this.config.project, this.config.env);
-			this.flagNodes = flags.map(flag => this.flagToValues(flag));
+			const nodes = [];
+			//await this.flagStore.isInitialized
+			const flags = await this.flagStore.allFlagsMetadata();
+			_.map(flags, value => {
```

Comment:
Any reason the array prototype `map` won't work here?

Reply:
`allFlagsMetadata()` returns `Promise<Dictionary<FeatureFlag>>`